### ISSUE TITLE
0.11 can not download blocks - on big endian machine

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [myriadteam/myriadcoin#32]
I. my problem (can't get the blocks downloaded)
II. what I investigated so far

**I. my problem :**
Compiling MYR works, but when I start it, 
I get always this in the debug.log:

  2016-09-29 11:28:28 ERROR: CheckProofOfWork(): hash doesn't match nBits (see I. below)
  2016-09-29 11:28:28 ERROR: CheckBlockHeader(): proof of work failed
  2016-09-29 11:28:28 ERROR: invalid header received

and soon after he bans the peers:
  Misbehaving: 108.61.10.90:10888 (50 -> 100) BAN THRESHOLD EXCEEDED

**II. what I investigated so far:**
Looking at Bitcoin, there are at least 2 merges for big endian:
M1: https://github.com/bitcoin/bitcoin/pull/5510 (2015 March)
M2: https://github.com/bitcoin/bitcoin/pull/7078/files (2015 Nov)

_note1:_
From what I've seen, Myriadcoin has integrated from merge M1 the changes for 14 of the 16 files

The changes in these 2 files are missing, but I can't find them in the current BTC code either:
a. src/primitives/block.cpp (has only 1 commit from 2016 Feb https://github.com/myriadteam/myriadcoin/commits/master/src/primitives/block.cpp)
b. src/txmempool.cpp (has 3 commits: 1 from 2014 and 2 from Feb 2016: https://github.com/myriadteam/myriadcoin/commits/master/src/txmempool.cpp)

_note2:_
Also merge M2 is missing in: src/uint256.h (https://github.com/myriadteam/myriadcoin/commits/master/src/uint256.h )

additional info:
version used: Myriad version v0.11.3.1-gd26f951 (Fri, 10 Jul 2015 19:23:55 +0200)
hardware: s390x IBM, tried both with OS: suse and debian

I. the error "hash doesn't match nBits" is inside pow.cpp:

```
// Check proof of work matches claimed amount
if (UintToArith256(hash) > bnTarget)
    return error("CheckProofOfWork(): hash doesn't match nBits");
```

**Comment by nzsquirrell** [comment]
pretty sure this is resolved in 0.14 tree, but have no big-endian architecture to test on.

**Comment by cryptapus** [comment]
0.11 will most likely not see further development
